File: sot-monitor/src/Model/Alert.php
<?php

namespace Ontic\Sot\Monitor\Model;

class Alert
{
    const PRIORITY_LOW = 0;
    const PRIORITY_NORMAL = 1;
    const PRIORITY_HIGH = 2;
    const PRIORITY_CRITICAL = 3;

    const TYPE_ACTIVATION = 'activation';
    const TYPE_DANGEROUS_ACTIVACION = 'dangerous_activation';
    const TYPE_INCONSISTENCY = 'inconsistency';

    /** @var string */
    private $type;
    /** @var array */
    private $data;
    /** @var int */
    private $timestamp;
    /** @var int */
    private $priority;

    public function __construct(string $type, array $data, int $timestamp, int $priority)
    {
        $this->type = $type;
        $this->data = $data;
        $this->timestamp = $timestamp;
        $this->priority = $priority;
    }

    public function getType(): string
    {
        return $this->type;
    }

    public function getData(): array
    {
        return $this->data;
    }

    public function getTimestamp(): int
    {
        return $this->timestamp;
    }

    public function getPriority(): int
    {
        return $this->priority;
    }
}

File: sot-android/app/src/main/java/es/ontic/sot/model/Alert.java
package es.ontic.sot.model;

import java.util.Map;

public class Alert
{
    public static final String TYPE_NEW_DEVICE = "new_device";
    public static final String TYPE_INTERMITTENT_POWER = "intermittent_power";
    public static final String TYPE_ACTIVATION = "activation";
    public static final String TYPE_DANGEROUS_ACTIVATION = "dangerous_activation";
    public static final String TYPE_INCOSISTENCY = "inconsistency";
    public static final String TYPE_DEVICE_OFFLLINE = "device_online";

    private String type;
    private Map<String, String> data;
    private long timestamp;
    private int priority;

    public Alert(String type, Map<String, String> data, long timestamp, int priority)
    {
        this.type = type;
        this.data = data;
        this.timestamp = timestamp;
        this.priority = priority;
    }

    public String getType()
    {
        return type;
    }

    public Map<String, String> getData()
    {
        return data;
    }

    public long getTimestamp()
    {
        return timestamp;
    }

    public int getPriority()
    {
        return priority;
    }

    public String getDevice()
    {
        return data.get("device");
    }
}
